src/store/middleware.js

```diff
@@ -66,6 +67,7 @@ export const stockMiddleware = store => next => action => {
       store.getState().chartData.selectedChartRange
     );
   } else if (action.type === UPDATE_CHART_RANGE) {
+    store.dispatch(resetChartDataAction());
```

Comment:
Why not just use the `UPDATE_CHART_RANGE` action type in the reducer to reset the, rather than making a new `RESET_CHART_DATA` action?

Reply:
Ohh yes! we can do it together. I didn't notice this before.